roboteam_ai/src/ApplicationManager.cpp

```diff
@@ -116,6 +116,7 @@ void ApplicationManager::updateDangerfinder() {
 void ApplicationManager::handleRefData() {
     ai::StrategyManager strategyManager;
     std::string strategyName = strategyManager.getCurrentStrategyName(refereeMsg.command);
+    BTFactory::getFactory().init();
```

Comment:
I suppose this rebuilds the trees. it looks like you reinitialize the trees every tick